spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClient.java

```diff
@@ -143,18 +143,10 @@ private List<ServiceInstance> getNamespaceServiceInstances(EndpointSubsetNS es,
 		Map<String, String> serviceMetadata = serviceMetadata(serviceId, service, properties, subsets, namespace);
 
 		for (EndpointSubset endpointSubset : subsets) {
-			List<EndpointAddress> addresses = endpointSubset.getAddresses();
```

Comment:
I've taken this one and moved it to the static utilities class, very slightly polished it and added tests